src/NativeScript/inspector/CachedResource.mm

```diff
@@ -0,0 +1,91 @@
+#include "CachedResource.h"
+#include "MimeTypeHelper.h"
+
+namespace Inspector {
+WTF::HashMap<WTF::String, Inspector::Protocol::Page::ResourceType> CachedResource::m_mimeTypeMap = {
+    { "text/xml", Inspector::Protocol::Page::ResourceType::Document },
+    { "text/plain", Inspector::Protocol::Page::ResourceType::Document },
+    { "application/xhtml+xml", Inspector::Protocol::Page::ResourceType::Document },
+    { "text/css", Inspector::Protocol::Page::ResourceType::Stylesheet },
+    { "text/javascript", Inspector::Protocol::Page::ResourceType::Script },
+    { "text/ecmascript", Inspector::Protocol::Page::ResourceType::Script },
+    { "application/javascript", Inspector::Protocol::Page::ResourceType::Script },
+    { "application/ecmascript", Inspector::Protocol::Page::ResourceType::Script },
+    { "application/x-javascript", Inspector::Protocol::Page::ResourceType::Script },
+    { "application/json", Inspector::Protocol::Page::ResourceType::Script },
+    { "application/x-json", Inspector::Protocol::Page::ResourceType::Script },
+    { "text/x-javascript", Inspector::Protocol::Page::ResourceType::Script },
+    { "text/x-json", Inspector::Protocol::Page::ResourceType::Script },
+    { "text/typescript", Inspector::Protocol::Page::ResourceType::Script },
+};
+
+CachedResource::CachedResource() {}
+
+CachedResource::CachedResource(WTF::String url)
+    : m_content(WTF::emptyString()) {
+    m_url = url;
+
+    NSString* cachedResourcePath = (NSString*)url;
+    NSURL* cachedResourceUrl = [NSURL URLWithString:cachedResourcePath];
+
+    m_mimeType = WTF::String(NativeScript::mimeTypeByExtension([cachedResourceUrl pathExtension]));
+    Inspector::Protocol::Page::ResourceType resourceType = Inspector::Protocol::Page::ResourceType::Other;
+    if (!m_mimeType.isEmpty()) {
+        resourceType = resourceTypeByMimeType(m_mimeType);
+    }
+
+    m_type = resourceType;
+}
+
+bool CachedResource::hasTextContent() {
+    return m_type == Inspector::Protocol::Page::ResourceType::Document || m_type == Inspector::Protocol::Page::ResourceType::Stylesheet || m_type == Inspector::Protocol::Page::ResourceType::Script || m_type == Inspector::Protocol::Page::ResourceType::XHR;
+}
+
+Inspector::Protocol::Page::ResourceType CachedResource::resourceTypeByMimeType(WTF::String mimeType) {
+    WTF::HashMap<WTF::String, Protocol::Page::ResourceType>::const_iterator iterator = m_mimeTypeMap.find(mimeType);
+    if (iterator != m_mimeTypeMap.end()) {
+        return iterator->value;
+    }
+
+    if (mimeType.startsWith("image/")) {
+        return Inspector::Protocol::Page::ResourceType::Image;
+    }
+
+    if (mimeType.startsWith("font/")) {
+        return Inspector::Protocol::Page::ResourceType::Font;
+    }
+
+    return Inspector::Protocol::Page::ResourceType::Other;
+}
+
+WTF::String CachedResource::content(ErrorString& out_error) {
+    if (m_content.isEmpty()) {
+        NSString* cachedResourcePath = (NSString*)m_url;
```

Comment:
Use `static_cast`, `reinterpret_cast` or `const_cast` where possible, refrain at all from C-style casts because they can strip `const` qualifiers and are potentially unsafe.